hat/assets/js/apps/Iaso/domains/forms/detail.js

```
@@ -0,0 +1,432 @@
+
+    const dispatch = useDispatch();
+    const intl = useSafeIntl();
+    const classes = useStyles();
+    const [
+        currentForm,
+        setFieldValue,
+        setFieldErrors,
+        setFormState,
+    ] = useFormState(initialFormState(initialData));
+
+    const onConfirm = async () => {
+        let isUpdate;
+        let saveForm;
+        let formData;
+        if (params.formId === '0') {
+            isUpdate = false;
+            formData = mapValues(
+                omit(currentForm, ['xls_file', 'form_id']),
+                v => v.value,
+            );
+            saveForm = createForm(dispatch, formData);
+        } else {
+            isUpdate = true;
+            formData = mapValues(omit(currentForm, 'xls_file'), v => v.value);
+            saveForm = updateForm(dispatch, currentForm.id.value, formData);
+        }
+        dispatch(setIsLoadingForm(true));
+        let isSaveSuccessful = false;
+        let savedFormData;
+        try {
+            savedFormData = await saveForm;
+            dispatch(setCurrentForm(savedFormData));
+            if (
+                !isUpdate ||
+                (isUpdate && currentForm.xls_file.value !== null)
+            ) {
+                try {
+                    await createFormVersion(
+                        dispatch,
+                        {
+                            form_id: savedFormData.id,
+                            xls_file: currentForm.xls_file.value,
+                        },
+                        isUpdate,
+                    );
+                    isSaveSuccessful = true;
+                } catch (createVersionError) {
+                    // when creating form, if version creation fails, delete freshly created, version-less form
+                    if (!isUpdate) {
+                        try {
+                            await deleteForm(dispatch, savedFormData.id);
+                            console.log('Form deleted');
```

What's your policy regarding console.log? Do you keep some of them in the codebase or is it for testing purposes?

---

nope, it was there before... i assume Pierre wanted to let it fail silently and keep a trace